src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/scatterPlot.js

```diff
@@ -1,154 +1,372 @@
     }
 
-    presentDataOnScatterplot(width, svg, height, colorDomain, colorRange) {
-        return function (_data) {
-            var headerNames = d3.keys(_data[0]);
+    addBrushing(box_width, box_height, scatter, scaleAndAxis, selectedZoomBtn, points, zoom) {
+        let extent;
+        let brush = d3.brush()
+            .extent([[0, 0], [box_width, box_height]])
+            .on("start brush", updateChart)
+
+        // The brush element must be child of zoom element - this is only way we found to have both zoom and brush
+        // events working at the same time. See https://stackoverflow.com/a/59757276 .
+        let brushElem = zoom.zoomElem.append("g")
+            .attr("class", "brush")
+            .call(brush)
+
+        let self = this;
+        function zoomIn() {
+            let xMin = scaleAndAxis.xScale.invert(extent[0][0]);
+            let xMax = scaleAndAxis.xScale.invert(extent[1][0]);
+            let yMin = scaleAndAxis.yScale.invert(extent[1][1]);
+            let yMax = scaleAndAxis.yScale.invert(extent[0][1]);
+
+            scaleAndAxis.xScale.domain([xMin, xMax]);
+            scaleAndAxis.yScale.domain([yMin, yMax]);
+
+            self.zoomingHelper(scaleAndAxis, box_width, scatter, points);
 
-            var x = d3.scaleLinear()
-                .domain([4, 8])
-                .range([0, width]);
-            var xAxis = svg.append("g")
-                .attr("transform", "translate(0," + height + ")")
-                .call(d3.axisBottom(x));
+            brushElem.call(brush.move, null);
+        }
 
-            var y = d3.scaleLinear()
-                .domain([0, 9])
-                .range([height, 0]);
-            svg.append("g")
-                .call(d3.axisLeft(y));
+        const zoomInKeyEvent = function (event) {
+            if ((event.ctrlKey || event.metaKey) && event.key === 'z') {
+                zoomIn();
+                selectedZoomBtn.style.display = "none";
+            }
+        };
 
-            var clip = svg.append("defs").append("svg:clipPath")
-                .attr("id", "clip")
-                .append("svg:rect")
-                .attr("width", width)
-                .attr("height", height)
-                .attr("x", 0)
-                .attr("y", 0);
+        function updateChart() {
+            let s = d3.event.selection;
+            selectedZoomBtn.style.display = "inline-block";
+            selectedZoomBtn.addEventListener("click",zoomIn,true)
+            document.addEventListener('keydown', zoomInKeyEvent,true);
+            extent = s;
+        }
 
-            var color = d3.scaleOrdinal()
-                .domain(colorDomain)
-                .range(colorRange)
+        const endBrushing = function (_) {
+            brushElem.call(brush.move, null);
+            selectedZoomBtn.style.display = "none";
+            selectedZoomBtn.removeEventListener("click",zoomIn,true)
+            document.removeEventListener('keydown', zoomInKeyEvent,true);
```

Comment:
all the styles in this file are inconsistent - sometimes there are spaces around `,`, sometimes not. Just apply Rust style guide here :)